mne/source_estimate.py

```diff
@@ -33,41 +33,42 @@
 
 def _read_stc(filename):
     """Aux Function."""
-    fid = open(filename, 'rb')
+    with open(filename, 'rb') as fid:
+        buf = fid.read()
 
     stc = dict()
-
-    fid.seek(0, 2)  # go to end of file
-    file_length = fid.tell()
-    fid.seek(0, 0)  # go to beginning of file
+    offset = 0
 
     # read tmin in ms
-    stc['tmin'] = float(np.fromfile(fid, dtype=">f4", count=1))
+    stc['tmin'] = float(np.frombuffer(buf, dtype=">f4", count=1, offset=offset))
     stc['tmin'] /= 1000.0
+    offset += np.dtype('>f4').itemsize
 
     # read sampling rate in ms
-    stc['tstep'] = float(np.fromfile(fid, dtype=">f4", count=1))
+    stc['tstep'] = float(np.frombuffer(buf, dtype=">f4", count=1, offset=offset))
     stc['tstep'] /= 1000.0
+    offset += np.dtype('>f4').itemsize
 
     # read number of vertices/sources
-    vertices_n = int(np.fromfile(fid, dtype=">u4", count=1))
+    vertices_n = int(np.frombuffer(buf, dtype=">u4", count=1, offset=offset))
+    offset += np.dtype('>u4').itemsize
 
     # read the source vector
-    stc['vertices'] = np.fromfile(fid, dtype=">u4", count=vertices_n)
+    stc['vertices'] = np.frombuffer(buf, dtype=">u4", count=vertices_n, offset=offset)
+    offset += np.dtype('>u4').itemsize * vertices_n
 
     # read the number of timepts
-    data_n = int(np.fromfile(fid, dtype=">u4", count=1))
+    data_n = int(np.frombuffer(buf, dtype=">u4", count=1, offset=offset))
+    offset += np.dtype('>u4').itemsize
```

Comment:
It's a bit silly to do this for every type. I would do ``n_bytes = 4`` and use that everywhere.